File: prueba4-front/src/app/services/entrenador.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baserUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class EntrenadorService {

  constructor(private http:HttpClient) { }

  public guardarEntrenador(entrenador:any){
    return this.http.post(`${baserUrl}/entrenador/`,entrenador);
  }

  public eliminarEntrenador(entrenadorId:any){
    return this.http.delete(`${baserUrl}/entrenador/${entrenadorId}`);
  }

  public actualizarEntrenador(entrenador:any){
    return this.http.put(`${baserUrl}/entrenador/`,entrenador);
  }

  public obtenerEntrenador(entrenadorId:any){
    return this.http.get(`${baserUrl}/entrenador/${entrenadorId}`);
  }

  public listarEntrenadores(){
    return this.http.get(`${baserUrl}/entrenadores/`);
  }

}

File: GetFit/tfg-master/prueba4-back/src/main/java/com/getfit/servicios/EntrenadorService.java
package com.getfit.servicios;

import java.util.Set;

import com.getfit.modelo.Entrenador;



public interface EntrenadorService {

	Set<Entrenador> obtenerEntrenadores();
	
	Entrenador agregarEntrenador(Entrenador entrenador);
	
	Entrenador actualizarEntrenador(Entrenador entrenador);
	
	Entrenador obtenerEntrenador(Long entrenadorId);
	
	void eliminarEntrenador(Long entrenadorId);
	
	Entrenador listarEntrenador(Long entrenadorId);
	
}
